test/partition-test/puml.cpp

```diff
@@ -0,0 +1,451 @@
+			std::default_random_engine gen(123);
+			std::uniform_int_distribution<int> distribution(1,1000);
+			for (int i = 0; i < puml->numOriginalCells(); ++i) {
+				vertexweights[i] = distribution(gen);
+			}
+		}
+		else if (vertexweightsetting == "increasing") {
+			vertexweights = new int[puml->cells().size()];
+			for (int i = 0; i < puml->numOriginalCells(); ++i) {
+				vertexweights[i] = i + 1;
+			}
+		}
+		else {
+			logInfo(rank) << "Vertex weight setting not recognized. Aborting...";
+			MPI_Finalize();
+			return 1;
+		}
+
+		if (edgeweightsetting == "none") {
+			// blank
+		}
+		else if (edgeweightsetting == "uniform") {
+			edgeweights = new int[graph.localEdgeCount()];
+			for (int i = 0; i < graph.localEdgeCount(); ++i) {
+				edgeweights[i] = 1;
+			}
+		}
+		else if (edgeweightsetting == "random") {
+			edgeweights = new int[graph.localEdgeCount()];
+			std::default_random_engine gen(123);
+			std::uniform_int_distribution<int> distribution(1,1000);
+			for (int i = 0; i < graph.localEdgeCount(); ++i) {
+				edgeweights[i] = distribution(gen);
+			}
+		}
+		else if (edgeweightsetting == "increasing") {
+			edgeweights = new int[graph.localEdgeCount()];
+			for (int i = 0; i < graph.localEdgeCount(); ++i) {
+				edgeweights[i] = i + 1;
+			}
+		}
+		else {
+			logInfo(rank) << "Edge weight setting not recognized. Aborting...";
+			MPI_Finalize();
+			return 1;
+		}
+
+		int nparts = readsetting<int>(settings, "nparts");
+
+		if (nodeweightsetting == "none") {
+			// empty
+		}
+		else if (nodeweightsetting == "uniform") {
+			nodeweights = new double[nparts];
+			for (int i = 0; i < nparts; ++i) {
+				nodeweights[i] = 1. / nparts;
+			}
+		}
+		else if (nodeweightsetting == "random") {
+			nodeweights = new double[nparts];
+			std::default_random_engine gen(123);
+			std::uniform_int_distribution<int> distribution(1,1000);
+			double sum = 0;
+			for (int i = 0; i < nparts; ++i) {
+				nodeweights[i] = distribution(gen);
+				sum += nodeweights[i];
+			}
+			for (int i = 0; i < nparts; ++i) {
+				nodeweights[i] /= sum;
+			}
+		}
+		else if (nodeweightsetting == "harmonic") {
+			nodeweights = new double[nparts];
+			double sum = 0;
+			for (int i = 0; i < nparts; ++i) {
+				sum += 1. / (i + 1);
+			}
+			for (int i = 0; i < nparts; ++i) {
+				nodeweights[i] = 1. / (sum * (i + 1));
+			}
+		}
+
+		graph.setVertexWeights(vertexweights, 1);
+		graph.set_edgeWeights(edgeweights);
+		double imbalance = readsetting<double>(settings, "imbalance");
+		int seed = readsetting<int>(settings, "seed");
+
+		PUML::PartitionTarget target;
+		target.setImbalance(imbalance);
+		if (nodeweights == nullptr) {
+			target.setVertexWeightsUniform(nparts);
+		}
+		else {
+			target.setVertexWeights(nparts, nodeweights);
+		}
+
+		logDebug() << "Initial stats" << rank << graph.localVertexCount() << graph.localEdgeCount();
+
+		logInfo(rank) << "Starting the actual partitioning.";
+		struct rusage startm;
```

Comment:
is the struct actually needed? startm could use a clearer name

Reply:
It tracks memory usage, was useful for testing back in 2019. It's renamed now (in the commit that'll be uploaded hopefully shortly).